iCount/analysis/rnamaps.py

```diff
@@ -3,690 +3,283 @@
+        nbins=50,
+        binsize=None,
+        colormap='Greys',
+        imgfmt='png',
+        ):
     """
     Compute distribution of cross-links relative to genomic landmarks.
 
     Parameters
     ----------
-    bam : str
-        BAM file with alligned reads.
-    segmentation : str
-        GTF file with segmentation. Should be a file produced by function
-        `get_segments`.
-    out_file : str
-        Output file with analysis results.
-    strange : str
-        File with strange propertieas obtained when processing bam file.
-    cross_transcript : str
-        File with reads spanning over multiple transcripts or multiple genes.
-    implicit_handling : str
-        Can be 'closest' or 'split'. In case of implicit read - split score to
-        both neighbours or give it just to the closest neighbour.
-    mismatches : int
-        Reads on same position with random barcode differing less than
-        ``mismatches`` are grouped together.
-    mapq_th : int
-        Ignore hits with MAPQ < mapq_th.
-    holesize_th : int
-        Raeads with size of holes less than holesize_th are treted as if they
-        would have no holes.
-    max_barcodes : int
-        Skip merging similar barcodes if number of distinct barcodes at
-        position is higher that this.
-
+    sites : str
+        Croslinks file (BED6 format). Should be sorted by coordinate.
+    regions : str
+        Regions file (regions.gtf.gz) that is produced by ``iCount segment``.
+    outdir : str
+        Output directory.
+    top_n : int
+        Plot heatmap for top_n best covered landmarks.
+    smoothing : int
+        Smoothing half-window. Average smoothing is used.
+    nbins : int
+        Number of bins. Either nbins or binsize can be defined, but not both.
+    binsize : int
+        Bin size. Either nbins or binsize can be defined, but not both.
+    colormap : str
+        Colormap to use. Any matplotlib colormap can be used.
+    imgfmt : str
+        OUtput image format.
 
     Returns
     -------
-    str
-        File with number of (al, explicit) scores per each position in each
-        RNA-map type.
+    None
 
     """
     iCount.logger.log_inputs(LOGGER)
 
-    if implicit_handling not in ('closest', 'split'):
-        raise ValueError(
-            'Parameter implicit_handling should be one of "closest" or "split"')
-
-    metrics = iCount.Metrics()
-    metrics.cross_transcript = 0
-    metrics.origin_premrna = 0
-    metrics.origin_mrna = 0
-    metrics.origin_ambiguous = 0
+    sites_name = iCount.files.remove_extension(sites, ['.bed', '.bed.gz'])
+    if outdir is None:
+        outdir = os.path.join(os.path.abspath(os.getcwd()), 'rnamaps_{}'.format(sites_name))
+        LOGGER.info('Output directory not given, creating one at %s', outdir)
+    os.makedirs(outdir, exist_ok=True)
+
+    for maptype in MAP_TYPES:
+        LOGGER.info('Creating landmarks for %s', maptype)
+        landmarks = make_landmarks_file(regions, maptype)
```

Comment:
The generation of the landmarks file should not happen the first time this analysis is run. This is a side-effect that is not to be expected (the rnamap function should generate rnamaps, and not landmarks). The file with landmarks should be a parameter of the function `run`, similarly as is the `regions` parameter.

The `segment` command should be modified to automatically generate the `landmarks` file(s).

If no `landmarks` file is provided when calling `run`, then `run` can check if the expected associated 'landmarks` file exists (as generated by `segment`). If not, it should fail.

We have to avoid side-effects of functions such as this. Also, the `regions` files may be somewhere where the user has no write rights. Also, generating temporary landmarks files every time the function is run, is probably not efficient and transparent. The user needs to see which landmarks are being used.